### PR TITLE
Make `run_with_accelerate` a pythonic decorator

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
This PR fixes the problem of calling `run_with_accelerate` as a python decorator directly on a function (e.g. `@run_with_accelerate...`) to make it usable with remote orchestrators like Skypilot.
As a side effect, it also forbids the functional use of this decorator, as this will not be reflected in the `Source` on remote orchestrator later on.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by schustmi — APPROVED** [review]

File: src/zenml/integrations/huggingface/steps/accelerate_runner.py [hunk 1]
```diff
@@ -65,100 +66,99 @@ def training_pipeline(some_param: int, ...):
...
+                "in a functional way with steps, please apply decoration "
+                "directly to a step instead. This behavior will be also "
+                "allowed in future, but now it faces technical limitations.\n"
+                "Example (allowed):\n"
+                f"@{run_with_accelerate.__name__}(...)\n"
+                f"def {step_function.name}(...):\n"
+                "    ...\n"
+                "Example (not allowed):\n"
+                "def my_pipeline(...):\n"
+                f"    run_with_accelerate({step_function.name},...)(...)\n"
```
Comment (bcdurak) [inline src/zenml/integrations/huggingface/steps/accelerate_runner.py:145]: I am not too familiar with `accelerate` myself, so I might be missing something here, but if this `RuntimeError` is true, we might want to change the docstring of the `run_with_accelerate` function, as it seems like the example that is given there is not allowed anymore.

Comment (avishniakov) [reply src/zenml/integrations/huggingface/steps/accelerate_runner.py:145]: Exactly the same is written on L143 🙂 `Example (not allowed):\n`

Comment (bcdurak) [reply src/zenml/integrations/huggingface/steps/accelerate_runner.py:145]: I was talking about L55 😄 

```python
@pipeline
def training_pipeline(some_param: int, ...):
    run_with_accelerate(training_step, num_processes=4)(some_param, ...)
```

Isn't this the not-allowed case?

Comment (avishniakov) [reply src/zenml/integrations/huggingface/steps/accelerate_runner.py:145]: Yeah, indeed. Good catch 👍🏼

**Pull request merged**